chains/solana/contracts/programs/ccip-router/src/instructions/v1/token_admin_registry.rs

```diff
@@ -53,20 +58,87 @@ pub fn register_token_admin_registry_via_owner(
 }
 
 pub fn set_pool(
-    ctx: Context<ModifyTokenAdminRegistry>,
+    ctx: Context<SetPoolTokenAdminRegistry>,
     mint: Pubkey,
-    pool_lookup_table: Pubkey,
+    writable_indexes: Vec<u8>,
 ) -> Result<()> {
+    // set new lookup table
     let token_admin_registry = &mut ctx.accounts.token_admin_registry;
     let previous_pool = token_admin_registry.lookup_table;
-    token_admin_registry.lookup_table = pool_lookup_table;
+    let new_pool = ctx.accounts.pool_lookuptable.key();
+    token_admin_registry.lookup_table = new_pool;
+
+    // set writable indexes
+    // used to build offchain tokenpool accounts in a transaction
+    // indexes can be checked to indicate is_writable for the specific account
+    // this is also used to validate to ensure the correct write permissions are used according the admin
+    token_admin_registry.reset_writable();
+    for ind in writable_indexes {
+        token_admin_registry.set_writable(ind)
+    }
 
-    // TODO: Validate here that the lookup table has everything
+    // validate lookup table contains minimum required accounts if not zero address
+    if new_pool != Pubkey::zeroed() {
+        // deserialize lookup table account
+        let lookup_table_data = &mut &ctx.accounts.pool_lookuptable.data.borrow()[..];
+        let lookup_table_account: AddressLookupTable =
+            AddressLookupTable::deserialize(lookup_table_data)
+                .map_err(|_| CcipRouterError::InvalidInputsLookupTableAccounts)?;
+        require!(
+            lookup_table_account.addresses.len() >= 9,
```

Comment:
Any chance you can link that `9` to a constant? It seems like that value is tied to logic in this file and in `pools.rs`, so it'd be nice if the compiler can check both places stay consistent in case we ever change one of them.

E.g., define a constant, use it here in the check, and use it in an explicit slice type in [this line](https://github.com/smartcontractkit/chainlink-ccip/pull/419/files#diff-47d7a8ac5ffa77e692d1f0db4fd1d601f82cec79acf45ba7fb67e036a0e3f0a1R183)
Said constant could be taken outside of the versioned logic folders if we consider that part of our interface with TokenPools.


Also, if you use `require_gte!` instead of `require!`, the logs will print the actual length in case of error, which can help when troubleshooting failed tests for example